Update customization docs (#188)
rancher/elemental-docs: docs/customizing.md

A common pattern is to combine the three ways described above. This pattern will allow you to add custom steps during the installation and add `cloud-init` files to be evaluated at boot time.

> Usually additional config files are added to the ISO by dynamically generating them at boot. This can be achieved by generating the ISO via a [SeedImage](/seedimage-reference.md) including custom cloud-config data.

Reviewer: ```suggestion
Additional config files can be added dynamically boot time by generating the ISO via a [SeedImage](/seedimage-reference.md) including custom cloud-config data.
```

> To apply this pattern, the ISO needs to include or generate at boot time:

Reviewer: ```suggestion
To apply this pattern, the following files need to be included in the ISO or generated at boot time:
```

1. A [configuration file](https://rancher.github.io/elemental-toolkit/docs/customizing/general_configuration/)

>    for the elemental client. The file must be named `config.yaml` and by default the elemental client looks for it under `/etc/elemental`, however
>    this path be configured as part of the installation parameters of a [MachineRegistration](/machineregistration-reference) resource.

Reviewer: ```suggestion
   for the elemental client. The file must be named `config.yaml` and located by default in `/etc/elemental`.
   This path can be configured as part of the installation parameters of a [MachineRegistration](/machineregistration-reference) resource.
```

#### Custom Elemental client configuration file

> [Elemental client](https://github.com/rancher/elemental-toolkit/blob/main/docs/elemental.md) `install`, `upgrade` and `reset` commands can be configured with a [custom configuration file](https://rancher.github.io/elemental-toolkit/docs/customizing/general_configuration/) located by default in `/etc/elemental/config.yaml` or, if you have multiple yaml files, the `/etc/elemental/config.d` directory will be loaded too.  

Reviewer: ```suggestion
[Elemental client](https://github.com/rancher/elemental-toolkit/blob/main/docs/elemental.md) `install`, `upgrade` and `reset` commands can be configured with a [custom configuration file](https://rancher.github.io/elemental-toolkit/docs/customizing/general_configuration/) located by default in `/etc/elemental/config.yaml`.
If you have multiple yaml files, you need to add them in the `/etc/elemental/config.d` directory.
```

cloud-init file. The local path is evaluated during
the installation, hence it must exists within the installation media, commonly an ISO image.

> By default, Elemental Teal live systems mount the ISO root at `/run/initramfs/live` and this could be the path set for `config-url` in `MachineRegistrations`:

Reviewer: ```suggestion
By default, Elemental Teal live systems mount the ISO root at `/run/initramfs/live` which is also the default path set for `config-url` in `MachineRegistrations`:
```

        - "/run/initramfs/live/oem/custom_config.yaml"
```
Elemental Teal live ISOs, when booted, have the ISO root mounted at `/run/initramfs/live`.
According to that, the ISO for the example above is expected to include the `/oem/custom_config.yaml` file.

> Note `/run/initramfs/live` is a readonly mountpoint, hence this not an appropriate path for dynamically generated content at ISO boot.

Reviewer: ```suggestion
:::note
`/run/initramfs/live` is a readonly mountpoint and it's not an appropriate path for dynamically generated content at ISO boot.
:::
```
